apps/api/src/movies/movies-search/movies-search.db.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@nestjs/common'
+
+import { PrismaService } from '../../database/prisma.service'
+
+import { MoviesSearchService } from './movies-search.service'
+
+interface MovieId {
+  movie_id: number
+}
+
+@Injectable()
+export class MoviesSearchDbService extends MoviesSearchService {
+  constructor (
+    private readonly db: PrismaService
+  ) {
+    super()
+  }
+
+  async findSimilarity (searchTitle: string, queryLimit: number = 10) {
+    return await this.db.$queryRaw<MovieId[]>`
+      SELECT movie_id FROM "movie_titles" m
+      WHERE ${searchTitle} % ANY(STRING_TO_ARRAY(m.title, ' '))
+      ORDER BY similarity(t.title, ${searchTitle}) DESC
+      LIMIT ${queryLimit};`
+  }
```

Comment:
You can use encapsulation in your favor to wrap "more complex" code. It simplifies testing a lot